File: lib/domain/public_plan/ports/public_plan_create_output_port.rb
# frozen_string_literal: true

module Domain
  module PublicPlan
    module Ports
      class PublicPlanCreateOutputPort
        def on_success(success_dto)
          raise NotImplementedError, "Subclasses must implement on_success"
        end

        def on_failure(failure_dto)
          raise NotImplementedError, "Subclasses must implement on_failure"
        end

        def on_no_crops_failure(view_context)
          raise NotImplementedError, "Subclasses must implement on_no_crops_failure"
        end
      end
    end
  end
end

File: crates/agrr-domain/src/public_plan/ports/public_plan_create_output_port.rs
use crate::public_plan::dtos::{
    PublicPlanCreateNoCropsViewContext, PublicPlanCreateOutput,
};
use crate::shared::dtos::Error;

/// Ruby: `Domain::PublicPlan::Ports::PublicPlanCreateOutputPort`
pub trait PublicPlanCreateOutputPort {
    fn on_success(&mut self, success_dto: PublicPlanCreateOutput);
    fn on_failure(&mut self, failure_dto: Error);
    fn on_no_crops_failure(&mut self, view_context: PublicPlanCreateNoCropsViewContext);
}
